hexa/databases/api.py

```diff
@@ -0,0 +1,92 @@
+import re
+
+import psycopg2
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from psycopg2 import sql
+from psycopg2.extensions import ISOLATION_LEVEL_AUTOCOMMIT
+
+
+def get_db_server_credentials():
+    return {
+        "role": settings.WORKSPACE_DATABASE_ROLE,
+        "password": settings.WORKSPACE_DATABASE_PASSWORD,
+        "host": settings.WORKSPACE_DATABASE_HOST,
+        "port": settings.WORKSPACE_DATABASE_PORT,
+    }
+
+
+def format_db_name(requested_name: str):
+    db_name = ""
+    for c in requested_name.lower().replace(" ", "_")[:31]:
```

Comment:
You have to check that the db starts with a letter or an underscore and the resulting db_name might be shorter than 31 characters if it includes unsupported chars. You need to limit the length after the processing.

Reply:
The check is done on a separate function (see  line 27)